cmd/execution_builder.go

```diff
@@ -620,14 +621,26 @@ func (exeNode *ExecutionNode) LoadExecutionDataGetter(node *NodeConfig) error {
 	return nil
 }
 
+func openChunkDataPackDB(dbPath string) (*badgerDB.DB, error) {
+	db, err := badgerDB.Open(badgerDB.LSMOnlyOptions(dbPath))
+	if err != nil {
+		return nil, fmt.Errorf("could not open chunk data pack badger db at path %v: %w", dbPath, err)
+	}
+	return db, nil
+}
+
 func (exeNode *ExecutionNode) LoadExecutionState(
 	node *NodeConfig,
 ) (
 	module.ReadyDoneAware,
 	error,
 ) {
 
-	chunkDataPacks := storage.NewChunkDataPacks(node.Metrics.Cache, node.DB, node.Storage.Collections, exeNode.exeConf.chunkDataPackCacheSize)
+	chunkDataPackDB, err := openChunkDataPackDB(exeNode.exeConf.chunkDataPackDir)
```

Comment:
given we're adding this mid-spork, what do you think about gating this behind a flag? Maybe if `--chunk-data-pack-dir` is set, use the new db, otherwise continue to use the protocol db.